File: motor3d 2.0/ecuacion.vb
Namespace Algebra
    Public Class Ecuacion
        Inherits ObjetoAlgebraico

        Private mVariables() As Double

        Public ReadOnly Property Variables() As Double()
            Get
                Return mVariables
            End Get
        End Property

        Public ReadOnly Property NumeroVariables() As Integer
            Get
                Return mVariables.GetUpperBound(0)
            End Get
        End Property

        Public ReadOnly Property TerminoIndependiente As Double
            Get
                Return mVariables(mVariables.GetUpperBound(0))
            End Get
        End Property

        Public Sub New(ByVal ParamArray ValVariables() As Double)
            mVariables = ValVariables
        End Sub

        Public Overrides Function ToString() As String
            Dim Retorno As String = ""
            For i As Integer = 0 To mVariables.GetUpperBound(0)
                If i = 0 Then
                    Retorno &= mVariables(i).ToString & LetraVariable(i)
                Else
                    If i < mVariables.GetUpperBound(0) Then
                        Retorno &= IIf(mVariables(i) >= 0, "+", "") & mVariables(i).ToString & LetraVariable(i)
                    Else
                        Retorno &= "=" & mVariables(i).ToString
                    End If
                End If
            Next

            Return Retorno
        End Function

        Public Function Copia() As Ecuacion
            Return Copia(Me)
        End Function

        Public Shared Function Copia(ByVal Ecuacion As Ecuacion) As Ecuacion
            Return New Ecuacion(Ecuacion.Variables)
        End Function

        Public Shared Function LetraVariable(ByVal IndiceVariable As Integer) As String
            Select Case IndiceVariable
                Case 0
                    Return "X"
                Case 1
                    Return "Y"
                Case 2
                    Return "Z"
                Case 3
                    Return "W"
                Case Else
                    Return "[VAR" & IndiceVariable & "]"
            End Select
        End Function
    End Class

End Namespace

File: motor3d 2.0.ConvertedToC#/Ecuacion.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Linq;
using System.Xml.Linq;
namespace Motor3D.Algebra
{
	public class Ecuacion : ObjetoAlgebraico
	{


		private double[] mVariables;
		public double[] Variables {
			get { return mVariables; }
		}

		public int NumeroVariables {
			get { return mVariables.GetUpperBound(0); }
		}

		public double TerminoIndependiente {
			get { return mVariables[mVariables.GetUpperBound(0)]; }
		}

		public Ecuacion(params double[] ValVariables)
		{
			mVariables = ValVariables;
		}

		public override string ToString()
		{
			string Retorno = "";
			for (int i = 0; i <= mVariables.GetUpperBound(0); i++) {
				if (i == 0) {
					Retorno += mVariables[i].ToString() + LetraVariable(i);
				} else {
					if (i < mVariables.GetUpperBound(0)) {
						Retorno += (mVariables[i] >= 0 ? "+" : "") + mVariables[i].ToString() + LetraVariable(i);
					} else {
						Retorno += "=" + mVariables[i].ToString();
					}
				}
			}

			return Retorno;
		}

		public Ecuacion Copia()
		{
			return Copia(this);
		}

		public static Ecuacion Copia(Ecuacion Ecuacion)
		{
			return new Ecuacion(Ecuacion.Variables);
		}

		public static string LetraVariable(int IndiceVariable)
		{
			switch (IndiceVariable) {
				case 0:
					return "X";
				case 1:
					return "Y";
				case 2:
					return "Z";
				case 3:
					return "W";
				default:
					return "[VAR" + IndiceVariable + "]";
			}
		}
	}

}
